docs/pages/router/reference/authentication.mdx

```diff
@@ -5,115 +5,352 @@ description: How to protect routes with Expo Router.
+      <Text
+        onPress={() => {
+          // The `app/(app)/_layout.tsx` will redirect to the sign-in screen.
+          signOut();
+        }}>
+        Sign Out
+      </Text>
     </View>
   );
 }
 ```
 
-Now if the authentication state changes globally, the user will be redirected to the appropriate route.
+We now have an app that can present a loading state while it checks the initial authentication state, and redirect to the sign-in screen if the user is not authenticated. If a user deep links to any protect routes, they'll be redirect back to the sign-in screen.
+
+## Alternative loading states
+
+In Expo Router, something must be rendered to the screen during the loading of the initial auth state. In the example above, we render a loading state in the app layout. Alternatively, you can make the `index` route a loading state, and move the initial route to something like `/home`, this is akin to how Twitter works.
+
+## Modals and per-route authentication
+
+Another common pattern is to render a sign-in modal over the top of the app. This enables you to dismiss partially preserve deep links when the authentication is complete. However, this pattern requires routes to be rendered in the background, these routes will need to handle data loading without authentication.
+
+<FileTree
+  files={[
+    ['app/_layout.js', 'Declares global session context'],
+    'app/(app)/_layout.js',
+    ['app/(app)/sign-in.js', <span>Modally presented over the root</span>],
+    ['app/(app)/(root)/_layout.js', <span>Protects child routes</span>],
+    [
+      'app/(app)/(root)/index.js',
+      <span>
+        Requires authorization <Lock01Icon className="inline mb-1" />
+      </span>,
+    ],
+  ]}
+/>
+
+```jsx app/(app)/_layout.js
+import { Stack } from 'expo-router';
+
+export const unstable_settings = {
+  initialRouteName: '(root)',
+};
+
+export default function AppLayout() {
+  return (
+    <Stack>
+      <Stack name="(root)" />
+      <Stack
+        name="sign-in"
+        options={{
+          presentation: 'modal',
+        }}
+      />
+    </Stack>
+  );
+}
+```
+
+## Navigating without navigation
+
+If you encounter the following error:
+
+```
+Error: Attempted to navigate before mounting the Root Layout component. Ensure the Root Layout component is rendering a Slot, or other navigator on the first render.
+```
+
+This can occur when the app attempts to perform navigation without a navigator mounted in the [root layout](/router/advanced/root-layout). You can fix this by adding a group and moving conditional logic up a level.
+
+### Before
+
+<FileTree files={['app/_layout.js', 'app/about.js']} />
+
+```js app/_layout.js
+export default function RootLayout() {
+  React.useEffect(() => {
+    // This navigation event will trigger the error above.
+    router.push('/about');
+  }, []);
+
+  // Avoid deferring the root layout, otherwise the navigation event will be triggered before the layout is mounted.
```

Comment:
```suggestion
  // This conditional statement creates a problem since the root layout's
  // content (the Slot) must be mounted before any navigation events occur.
```